src/main/scala/cromwell/engine/SymbolStore.scala

```diff
@@ -23,7 +24,7 @@ object SymbolStoreEntry {
 case class SymbolStoreEntry(key: SymbolStoreKey, wdlType: WdlType, wdlValue: Option[WdlValue]) {
 
   wdlValue match {
-    case Some (value) => wdlType.checkCompatible(value.value)
+    case Some (value) => if(wdlType != value.wdlType) throw new UnsupportedOperationException("Fix this.")
```

Comment:
This pattern shows up a lot. It raises a couple of questions for me:

* Can these mismatches actually happen? By that I mean if we the developers don't screw up can the bad state ever occur? If not I think we shouldn't be overly defensive.
* Is there any way we can enforce these via type system higher up in the chain?

I should have probably done more homework on both of these before asking but I didn't :)